modules/terraform/count_test.go

```diff
@@ -0,0 +1,95 @@
+package terraform
+
+import (
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/files"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetResourceCount(t *testing.T) {
+	t.Parallel()
+	testFolder, err := files.CopyTerraformFolderToTemp("../../test/fixtures/terraform-basic-configuration", t.Name())
+	require.NoError(t, err)
+
+	terraformOptions := &Options{
+		TerraformDir: testFolder,
+		Vars: map[string]interface{}{
+			"cnt": 1,
+		},
+	}
+
+	cnt := GetResourceCount(t, InitAndPlan(t, terraformOptions))
+	assert.Equal(t, 1, cnt.Add)
+	assert.Equal(t, 0, cnt.Change)
+	assert.Equal(t, 0, cnt.Destroy)
+}
+
+func TestGetResourceCountEColor(t *testing.T) {
+	t.Parallel()
+	runTestGetResourceCountE(t, false)
+}
+
+func TestGetResourceCountENoColor(t *testing.T) {
+	t.Parallel()
+	runTestGetResourceCountE(t, true)
+}
+
+func runTestGetResourceCountE(t *testing.T, noColor bool) {
+	testCases := []struct {
+		Name                                         string
+		tfFuncToRun                                  func(t *testing.T, options *Options) string
+		cntValue                                     int
+		expectedAdd, expectedChange, expectedDestroy int
+	}{
+		{"Plan_Zero", InitAndPlan, 0, 0, 0, 0},
+		{"Apply_Zero", InitAndApply, 0, 0, 0, 0},
+		{"Plan_AddResouce", InitAndPlan, 2, 2, 0, 0},
+		{"Apply_AddResouce", InitAndApply, 2, 2, 0, 0},
+		{"Plan_NoOp", InitAndApply, 2, 0, 0, 0},
+		{"Apply_NoOp", InitAndApply, 2, 0, 0, 0},
+		{"Plan_DestroyResource", InitAndPlan, 1, 0, 0, 1},
+		{"Apply_DestroyResource", InitAndApply, 1, 0, 0, 1},
+		{"Destroy", Destroy, 1, 0, 0, 1},
+		{"Destroy_NoOp", Destroy, 1, 0, 0, 0},
+	}
+
+	notFailedSoFar := true
+
+	testFolder, err := files.CopyTerraformFolderToTemp("../../test/fixtures/terraform-basic-configuration", t.Name())
+	require.NoError(t, err)
+
+	terraformOptions := &Options{
+		TerraformDir: testFolder,
+		Vars: map[string]interface{}{
+			"cnt": 0,
+		},
+		NoColor: noColor,
+	}
+
+	for _, tc := range testCases {
+		if notFailedSoFar {
```

Comment:
Is this guard necessary? It doesn't look like a failure in one subtest will affect the others, so this is probably not necessary (and complicates the test: took me a bit to understand what was going on).

As a side note, if you did indeed want this behavior where a failure in a subtest should short circuit the test, then you shouldn't use subtests. Then, you can switch `assert` with `require` to get that behavior. The idea of subtests are that each unit is self contained.

Reply:
agree, this `notFailedSoFar` introduces misunderstanding, removed. The purpose of that construction was  to mimic `-failfast` on the level of subtests: failed test 1) runs until its end (`assert`), 2) aborts execution of all subsequent tests.